testsuite/src/main/java/io/netty/testsuite/transport/socket/AbstractDatagramTest.java

```diff
@@ -41,16 +39,11 @@ protected AbstractDatagramTest() {
 
     @Override
     protected void configure(Bootstrap bootstrap, Bootstrap bootstrap2, ByteBufAllocator allocator) {
-        addr = new InetSocketAddress(
-                NetUtil.LOCALHOST4, TestUtils.getFreePort());
-        bootstrap.localAddress(addr);
         bootstrap.option(ChannelOption.ALLOCATOR, allocator);
-        bootstrap2.localAddress(0).remoteAddress(addr);
         bootstrap2.option(ChannelOption.ALLOCATOR, allocator);
     }
 
-    protected void refreshLocalAddress(Bootstrap bootstrap) {
-        addr = new InetSocketAddress(NetUtil.LOCALHOST4, TestUtils.getFreePort());
-        bootstrap.localAddress(addr);
+    protected SocketAddress newSocketAddress() {
+        return new InetSocketAddress(NetUtil.LOCALHOST4, 0);
```

Comment:
it looks like the "loop 3 times" has been removed in `DatagramUnicastTest.java` ... do we still have to force ipv4?